src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -2619,6 +2619,11 @@ private static bool IsEquivalentSymbol(ISymbol declaredSymbol, ISymbol otherSymb
                                 operationsToAnalyze.Add(operationBlock.Parent);
                                 break;
 
+                            case OperationKind.ExpressionStatement:
+                                // For constructor initializer, we generate an IInvocationOperation with an implicit IExpressionStatementOperation parent.
```

Comment:
>For constructor initializer, we generate an IInvocationOperation with an implicit IExpressionStatementOperation parent. [](start = 35, length = 119)

The assert looks less specific. Why don't we assert exactly what we expect, i.e.:
- the parent is implicit IExpressionStatementOperation
- the parent.Parent is IConstructorBodyOperation
- the parent is in initializer position within parent.Parent

Reply:
Thanks, done.